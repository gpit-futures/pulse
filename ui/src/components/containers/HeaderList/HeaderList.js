import React, { PureComponent } from 'react';

export default class HeaderList extends PureComponent {
  static defaultProps = {
    items: [],
  };

  render() {
    const { items } = this.props;
    return (
      (items && items.length)
        ? <div className="header-list">
          { items.map((el, i) => (<div className="header-list__item" key={i}>{ el }</div>)) }
        </div>
        : <div />
    )
  }
}
