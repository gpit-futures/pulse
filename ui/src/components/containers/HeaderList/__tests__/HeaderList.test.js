import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';

import HeaderList from '../HeaderList';

Enzyme.configure({ adapter: new Adapter() });

const testProps = {
  items: [
    <img src="test1" alt="header img 1" />,
    <img src="test2" alt="header img 2" />,
  ],
};

describe('Component <HeaderList />', () => {
  it('should renders correctly when items is not Empty', () => {
    const component = shallow(
      <HeaderList
        items={testProps.items}
      />);

    expect(component.find('.header-list')).toHaveLength(1);
    expect(component.find('img')).toHaveLength(2);

    expect(component).toMatchSnapshot();
  });

  it('should renders correctly when items is Empty', () => {
    const component = shallow(
      <HeaderList
        items={[]}
      />);

    expect(component.find('.header-list')).toHaveLength(0);
    expect(component.find('img')).toHaveLength(0);

    expect(component).toMatchSnapshot();
  });
});

