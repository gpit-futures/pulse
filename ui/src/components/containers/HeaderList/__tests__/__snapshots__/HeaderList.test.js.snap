// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <HeaderList /> should renders correctly when items is Empty 1`] = `<div />`;

exports[`Component <HeaderList /> should renders correctly when items is not Empty 1`] = `
<div
  className="header-list"
>
  <div
    className="header-list__item"
    key="0"
  >
    <img
      alt="header img 1"
      src="test1"
    />
  </div>
  <div
    className="header-list__item"
    key="1"
  >
    <img
      alt="header img 2"
      src="test2"
    />
  </div>
</div>
`;
